targets/targets.json

```diff
@@ -2826,6 +2826,29 @@
             "network-default-interface-type": "ETHERNET"
         }
     },
+    "MTB_STM_S2LP": {
+        "inherits": ["FAMILY_STM32"],
+        "core": "Cortex-M4F",
+        "config": {
+            "clock_source": {
+                "help": "Mask value : USE_PLL_HSE_EXTC | USE_PLL_HSE_XTAL (8MHz) | USE_PLL_HSI",
+                "value": "USE_PLL_HSE_XTAL",
+                "macro_name": "CLOCK_SOURCE"
+            },
+            "clock_source_usb": {
+                "help": "As 48 Mhz clock is configured for USB, SYSCLK has to be reduced from 180 to 168 MHz (set 0 for the max SYSCLK value)",
+                "value": "1",
+                "macro_name": "CLOCK_SOURCE_USB"
+            }
+        },
+        "extra_labels_add": ["STM32F4", "STM32F429", "STM32F429ZI", "STM32F429xx", "STM32F429xI"],
+        "macros_add": ["USB_STM_HAL"],
+        "device_has_add": ["ANALOGOUT", "CAN", "SERIAL_ASYNCH", "TRNG", "FLASH", "MPU"],
```

Comment:
If this MTB has S2LP radio, and the driver is already in Mbed OS https://github.com/ARMmbed/mbed-os/tree/master/components/802.15.4_RF/stm-s2lp-rf-driver

We should enable it by default, so `802_15_4_PHY` should be in the list, also Default network interface type should be MESH.

So add:
```
        "overrides": {
            "network-default-interface-type": "MESH"
        }
```

Then, within the driver, enable it by default:
https://github.com/ARMmbed/mbed-os/blob/master/components/802.15.4_RF/stm-s2lp-rf-driver/mbed_lib.json#L8

Add
```
    "target_overrides": {
        "MTB_STM_S2LP": {
           "s2lp.provide-default": true
        }
    }
```

Reply:
@JarkkoPaso : Will you take up the change @SeppoTakalo suggested for the driver? I will make the change for the default network interface. Thanks.

Reply:
@ashok-rao Yes, I can make the driver change.

Reply:
Do the changes in this same PR so that it contains the complete work. If you do a separate PR regarding the driver work, it will be broken until both are merged.

Reply:
Done.

Reply:
I still don't see the `802_15_4_PHY` in the `device_has.add` list.

Otherwise looks OK.